fedimint-client/src/api_announcements.rs

```diff
@@ -37,81 +38,111 @@ impl_db_lookup!(
-                    };
-
-                    if !announcement.verify(SECP256K1, guardian_pub_key) {
-                        bail!("Failed to verify announcement for peer {}", peer_id);
-                    }
-                }
+        let _ =
+            refresh_api_announcement_sync(&client_inner.api, client_inner.db(), &guardian_pub_keys)
+                .await;
+
+        // Check once an hour if there are new announcements
+        sleep(Duration::from_secs(3600)).await;
+    }
+}
 
-                client_inner
-                    .db()
-                    .autocommit(
-                        |dbtx, _|{
-                            let announcements_inner = announcements.clone();
-                        Box::pin(async move {
-                            for (peer, new_announcement) in announcements_inner {
-                                let replace_current_announcement = dbtx
-                                    .get_value(&ApiAnnouncementKey(peer))
-                                    .await.is_none_or(|current_announcement| {
-                                        current_announcement.api_announcement.nonce
-                                            < new_announcement.api_announcement.nonce
-                                    });
-                                if replace_current_announcement {
-                                    debug!(target: LOG_CLIENT, ?peer, %new_announcement.api_announcement.api_url, "Updating API announcement");
-                                    dbtx.insert_entry(&ApiAnnouncementKey(peer), &new_announcement)
-                                        .await;
-                                }
-                            }
-
-                            Result::<(), ()>::Ok(())
-                        })},
-                        None,
-                    )
-                    .await
-                    .expect("Will never return an error");
-
-                Ok(())
-            })).await;
-
-        for (peer_id, result) in guardian_pub_keys.keys().zip(results) {
-            if let Err(e) = result {
+pub(crate) async fn refresh_api_announcement_sync(
+    api: &DynGlobalApi,
+    db: &Database,
+    guardian_pub_keys: &BTreeMap<PeerId, bitcoin::secp256k1::PublicKey>,
+) -> anyhow::Result<()> {
+    let results = fetch_api_announcements_from_all_peers(api, guardian_pub_keys).await;
+
+    let mut some_success = false;
+
+    for (peer_id, result) in guardian_pub_keys.keys().zip(results) {
+        match result {
+            Ok(announcements) => {
+                store_api_announcements(db, announcements).await;
+                some_success |= true
+            }
+            Err(e) => {
                 warn!(target: LOG_CLIENT, %peer_id, ?e, "Failed to process API announcements");
             }
         }
+    }
 
-        // Check once an hour if there are new announcements
-        sleep(Duration::from_secs(3600)).await;
+    if some_success {
```

Comment:
If we at least got success from a single peer, we should consider that enough (success).